services/user_interface/main.py

```diff
@@ -127,6 +128,62 @@ def user_labels():
     print(r.json())
     return '{}'
 
+@app.route('/get_topics_and_news')
+def get_topics_and_news():
+    """
+    use this endpoint to get news topics.
+
+    e.g.: http://localhost:5000/get_topics_and_news?n=5&bias=left
+    :return:
+    """
+    n_topics = int(request.args.get('n', default=5))
+    bias = request.args.get('bias', default="")
+    topics = requests.get(
+        'http://news_crawler:{}/get_topics?n={}'.format(
+            NEWS_CRAWLER_HTTP_PORT,
+            n_topics
+        )
+    ).json()
+
+    news = requests.get(
+        'http://news_crawler:{}/get_news?bias={}'.format(
+            NEWS_CRAWLER_HTTP_PORT,
+            request.args.get('bias', default="")
+        )
+    ).json()
+
+    for topic_idx, topic in enumerate(topics):
+        topic['articles'] = [ article for article in news
+                                if article['topic'] == topic_idx ]
+
+    return json.dumps(topics)
+
+@app.route('/get_topics')
+def get_topics():
+    """
+    use this endpoint to get news topics.
+
+    e.g.: http://localhost:5000/get_topics?n=5
+    :return:
+    """
+    n_topics = int(request.args.get('n', default=5))
+    r = requests.get(
+        'http://news_crawler:{}/get_topics?n={}'.format(
+            NEWS_CRAWLER_HTTP_PORT,
+            n_topics
+        )
+    )
+    return json.dumps(r.json())
+
+@app.route('/get_news')
+def get_news():
+    r = requests.get(
+        'http://news_crawler:{}/get_news?bias={}'.format(
```

Comment:
und gibt es auch den endpoint beim news crawler?

Reply:
sorry, hatte vergessen einzuchecken, ja gibts und hat auch funktioniert. sieht noch scheisse aus, und die linksrechts einordnung is auch noch nich so super, kommt viel muell. aber geht erstma